src/core/regimes.rkt

```diff
@@ -41,10 +41,13 @@
            (write (option-splitpoints opt) port)
            (display ">" port))])
 
-;; TODO: Figure out alts
 (define (exprs-to-branch-on alts)
-  (define critexprs (all-critical-subexpressions (*start-prog*)))
-  (remove-duplicates critexprs))
+  (if (flag-set? 'reduce 'branch-expressions)
+      (let ([alt-critexprs (for/list ([alt alts])
+              (all-critical-subexpressions (alt-program alt)))]
+            [critexprs (all-critical-subexpressions (*start-prog*))])
+           (remove-duplicates (foldr append '() (cons critexprs alt-critexprs))))
+      (free-variables (location-get '(1) (*start-prog*)))))
```

Comment:
Hmm—I'm not sure this is right. `'(1)` on the start program gets the list of program variables, like `(a b c)`; the free variables of that as an expression should be all but the first variable, in this case `b` and `c`. In any case, you can use the `program-variables` function to get the variables for a program.